builtin/providers/aws/resource_aws_nat_gateway.go

```diff
@@ -101,7 +101,14 @@ func resourceAwsNatGatewayRead(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		return err
 	}
-	if ngRaw == nil || strings.ToLower(state) == "deleted" {
+
+	status := map[string]bool{
+		"deleted":  true,
+		"deleting": true,
+		"failed":   true,
+	}
+
+	if ngRaw == nil || status[strings.ToLower(state)] {
```

Comment:
I would suggest checking for key existence the Go way, for example:

`if _, ok := status[strings.ToLower(state)]; ngRaw == nil || ok {`